flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/fetcher/SplitFetcherTest.java

```diff
@@ -275,6 +279,37 @@ public void testCloseAfterPause() throws InterruptedException {
         assertThat(fetcher.runOnce()).isFalse();
     }
 
+    @Test
+    public void testShutdownWaitingForRecordsProcessing() throws Exception {
+        TestingSplitReader<Object, TestingSourceSplit> splitReader = new TestingSplitReader<>();
+        FutureCompletingBlockingQueue<RecordsWithSplitIds<Object>> queue =
+                new FutureCompletingBlockingQueue<>();
+        final SplitFetcher<Object, TestingSourceSplit> fetcher = createFetcher(splitReader, queue);
+        fetcher.shutdown(true);
+
+        // Spawn a new thread af fetcher thread to go through the shutdown sequence.
+        CheckedThread fetcherThread =
+                new CheckedThread() {
+                    @Override
+                    public void go() throws Exception {
+                        fetcher.run();
+                        assertThat(splitReader.isClosed()).isTrue();
+                    }
+                };
+        fetcherThread.start();
+
+        // Wait until the fetcher thread to block on the shutdown latch.
+        waitUntil(
+                () -> fetcherThread.getState() == WAITING,
+                Duration.ofSeconds(1),
+                "The fetcher thread should be waiting for the shutdown latch");
+        assertFalse("The split reader should have not been closed.", splitReader.isClosed());
```

Comment:
The latest code style requires AssertJ for tests:

```suggestion
        assertThat(splitReader.isClosed()).as("The split reader should have not been closed.").isFalse();
```